rollup.config.js

```diff
@@ -1,27 +1,97 @@
+import pkg from './package.json';
 import commonjs from 'rollup-plugin-commonjs';
+import nodeBuiltins from 'rollup-plugin-node-builtins';
 import nodeResolve from 'rollup-plugin-node-resolve';
 import babel from 'rollup-plugin-babel';
 
-export default {
-  input: 'index.js',
-  output: {
-    file: 'dist/mongodb-extjson.js',
-    format: 'umd',
-    name: 'mongodb-extjson',
-  },
-  plugins: [
-    nodeResolve(),
-    commonjs(),
-    babel({
-      presets: [
-        [
-          'env',
-          {
-            modules: false
-          }
-        ]
+const input = 'index.js';
+const plugins = [
+  nodeResolve({
+    preferBuiltins: true
+  }),
+  commonjs(),
+  nodeBuiltins(),
+  babel({
+    plugins: [ 'external-helpers' ],
+    presets: [
+      [
+        'env',
+        {
+          modules: false
+        }
+      ]
+    ]
+  })
+];
+const browserPlugins = [
+  nodeResolve({
+    browser: true,
+    preferBuiltins: false
+  }),
+  commonjs(),
+  nodeBuiltins(),
+  babel({
+    plugins: [ 'external-helpers' ],
+    presets: [
+      [
+        'env',
+        {
+          modules: false
+        }
       ]
-    })
-  ],
-  external: ['bson'],
-}
+    ]
+  })
+];
+const external = Object.keys(pkg.dependencies || {});
+const defaultName = 'mongodb-extjson';
```

Comment:
you mentioned this the other day, but I think we should standardize on calling this module `ejson`. Code examples should already be using `EJSON.parse`/`EJSON.stringify`, similar to the javascript `JSON` module. If you want to throw that into this PR I'd be fine with that

Reply:
Discussed offline. Can only change this once we are okay with going to 4.x.